### ISSUE TITLE
Power is set to zero after having series of values

### ISSUE THREAD

**Issue by nirozjung** [jfarcher/energymonitor#1]
Hi, i tried the code to use for a simple run time power calculation without use of mqtt,  but power is all of sudden returned to zero. What I did is just removed all the mqtt parts of the code as I opted to use serial interface with pi.  
Is true that power is set to 0 for 1 minute during the startup by the following code. 
if (millis() - chan1_last_pulse > 60000) {
chan1_watts = 0;
}

**Comment by jfarcher** [comment]
so are you outputting to serial from the arduino to the raspberry pi instead of MQTT? are you getting any serial output at all?

**Comment by nirozjung** [comment]
Hi Jon, 
Thanks for the response. Yes, I am outputting to serial and checking the serial data using minicom. Actually I want to parse these data to openhab. But I am recieving bunch of zeroes as output, and whenever i unplug and plug ground, it gives series of power values and it ceases back to ground even if there is load is ON. I am using ABB c11 single phase meter with 100imp/1kWhr. I am also outputing the energy values at each time when the counter increases. I am confused if it right way. I changed the code as follows. 

const int chan1_pin = 2;
//My meter flashes 100 times per kWh
const float w_per_pulse = 10;
const unsigned long ms_per_hour = 3600000UL;

unsigned int chan1_count = 0;
unsigned long report_time = 0;
unsigned long chan1_first_pulse = 0;
unsigned long chan1_last_pulse = 0;
volatile byte chan1_pulse = 0;
unsigned long WH=0;

void chan1_isr() {
  chan1_pulse = 1;
}

void setup() {
  pinMode(chan1_pin, INPUT);
  attachInterrupt(0, chan1_isr, RISING);
  Serial.begin(9600);
  delay(1000);
}

void loop() {
  static unsigned int chan1_watts;
  unsigned long chan1_delta; // Time since last pulse
  if (millis() - chan1_last_pulse > 60000) {
    chan1_watts = 0;
  }
  if (millis() - report_time > 5000) {
    chan1_delta = chan1_last_pulse - chan1_first_pulse;
    if (chan1_delta && chan1_count) {
      chan1_watts = (chan1_count - 1) \* w_per_pulse \* ms_per_hour / chan1_delta;
      chan1_count = 0;
    }
    if (!(report_time == 0 && chan1_watts == 0)) {
      Serial.print("Ardunio_power:");
      Serial.println(chan1_watts);
      report_time = millis();
    }
  }
  if (chan1_pulse == 1) {
    chan1_count++;
    WH=chan1_count*10;
    Serial.print("Arduino_energy");
    Serial.println(WH);
    chan1_pulse = 0;

```
chan1_last_pulse = millis();
if (chan1_count == 1) { // was reset
  chan1_first_pulse = chan1_last_pulse;
} // end inner if
```

  } // end if
  } // end loop

I am confused, if it is the way I am doing right. 
Regards,

**Comment by jfarcher** [comment]
No worries, I'm intrigued.
Haven't had a chance to have a look at your code modifications properly but I will. However, you do know that OpenHab can be configured to recieve MQTT?

**Comment by nirozjung** [comment]
Yeah Jon, I know that well and works perfect but for that i need to get some extra hardwares( I guess Ethernet Shield one) if I am right.

**Comment by jfarcher** [comment]
This is true, sorry I didn't think about the extra hardware aspect... That being said I can definitely recommend an Ethernet Shield.
